### PR TITLE
Add support for specifying the SSG configuration file path

### LINKED ISSUE #5: Support for specifying folder
My next project is in a monorepo, and therefore the `next.config.js` isn't on root.

An input so I could specify the working directory would solve this for me.

### PR DESCRIPTION
**Pull request by JamesMGreene**
Fixes #5

Example usage:

```yaml
    steps:
      - name: Setup Pages
        id: pages
        uses: actions/configure-pages@ssg-config-file-input
        with:
          static_site_generator: next
          generator_config_file: some/relative/or/absolute/path/next.config.js
```

I also added support for updating ESM syntax configuration files that do not have object literal expressions in the `export default` declaration.

**Known Issue:** If the specified configuration file does not yet exist, we may end up writing the incorrect syntax (ESM vs. CommonJS, and vice versa) when creating the file.

### CODE REVIEW

**Comment by JamesMGreene** [comment]
cc @karl-run if you'd like to test this out 👍🏻

File: dist/index.js [hunk 1]
```diff
@@ -14584,14 +14584,32 @@ class ConfigParser {
...
+
+    // Direct default export
+    if (defaultExport && defaultExport.declaration.type === 'ObjectExpression') {
+      core.info('Found configuration object in direct default export declaration')
       return defaultExport.declaration
     }
 
+    // Indirect default export
+    else if (defaultExport && defaultExport.declaration.type === 'Identifier') {
+      const identifierName = defaultExport && defaultExport.declaration.name
```
Comment (yoannchaudet) [inline dist/index.js:14597]: `defaultExport` is already defined in this block

File: src/config-parser.js [hunk 2]
```diff
@@ -56,14 +56,32 @@ class ConfigParser {
...
+
+    // Direct default export
+    if (defaultExport && defaultExport.declaration.type === 'ObjectExpression') {
+      core.info('Found configuration object in direct default export declaration')
       return defaultExport.declaration
     }
 
+    // Indirect default export
+    else if (defaultExport && defaultExport.declaration.type === 'Identifier') {
+      const identifierName = defaultExport && defaultExport.declaration.name
```
Comment (yoannchaudet) [inline src/config-parser.js:69]: `defaultExport` is already defined in this block

```suggestion
      const identifierName = defaultExport.declaration.name
```

**Review by yoannchaudet — APPROVED** [review]
Looking good!

**Review by yoannchaudet — APPROVED** [review]
Looking good!

**Comment by JamesMGreene** [comment]
Testing looks good (after the fix in https://github.com/actions/configure-pages/pull/15/commits/fad78054b67df4fb9c28d3e8cb739100c39affdd) in an example repo with some alternate Gatsby config file names. 👍🏻

**Pull request merged**